test/testinfra.py

```diff
@@ -325,26 +322,32 @@ def update_status(revision, context, last, changes):
                 results.append((priority, "master", revision, "master", context))
                 update_status(revision, context, status, changes)
 
-        if pull_contexts:
-            pulls = self.get("pulls")
-            for pull in pulls:
+        pulls = self.get("pulls")
+        for pull in pulls:
+            number = pull["number"]
+            labels = self.labels(number)
+            revision = pull["head"]["sha"]
+            statuses = self.statuses(revision)
+            login = pull["head"]["user"]["login"]
+
+            for context in contexts.keys():
+                status = statuses.get(context, None)
                 baseline = 10
 
-                # It needs to be in the whitelist
-                login = pull["head"]["user"]["login"]
-                if login not in whitelist:
-                    baseline = 0
+                # Only create new status for those requested
+                if not status:
+                    if context not in pull_contexts:
+                        continue
+                    status = { }
 
-                number = pull["number"]
-                labels = self.labels(number)
-                revision = pull["head"]["sha"]
-                statuses = self.statuses(revision)
+                # For unmarked untested status, user must be in whitelist
+                if login not in self.whitelist:
+                    if status.get("description", NO_TESTING) == NO_TESTING:
```

Comment:
That's actually a likely misunderstanding, so I should add a comment here.

It's only for the given commit. Since statuses are per commit. Once the user pushes a new commit into the pull request, it must be triggered for testing again. It's not whitelisting of the login or pull request.